charts/consul/templates/client-snapshot-agent-deployment.yaml

```diff
@@ -137,10 +147,17 @@ spec:
           {{- .Values.client.snapshotAgent.caCert | nindent 14 }}
           EOF
           {{- end }}
+          {{- if  .Values.global.secretsBackend.vault.enabled }}
+          cat /vault/secrets/snapshot-agent-config.txt | base64 -d > /vault/secrets/snapshot-agent-config.json
```

Comment:
I don't think that `base64 -d` is platform agnostic. I am sure it is `base64 -D` on linux. Who knows what Windows is.

It might be possible to use the built in helm functions to do this. The function is something like `b64enc`. If the vault file is already a json encode, can you get as snapshot-agent-config.json and do something like:

{{ "/vault/secrets/snapshot-agent-config.json" | b64enc }}

I am not 100% sure that works.

Reply:
great call!  🙏

Reply:
I tried a few stabs at this and could never get this working and found a solution that achieves what we want.  We can use consul template's base64 decode in the vault agent injector annotation and have it write directly to a .json file that is decode.  So there is no need to copy from a .txt file to a .json file.  It's in my latest commit.